lib/porky_lib/file_service_helper.rb

```diff
@@ -56,6 +63,12 @@ def generate_file_key(options)
   private
 
   def a_file?(file_or_content)
-    !file_or_content.is_a?(String)
+    file_or_content.is_a?(File) || file_or_content.is_a?(Tempfile)
+  end
+
+  def a_path?(content_or_path)
+    return false if content_or_path.include?("\u0000")
```

Comment:
how did you create that file with a null byte? I wonder if the position for that null byte affected these measures. 
My concern was that this could have been a very expensive operation for big files, but it doesn't look too bad.

Also, did you test this with 75GB files? I think that's the limit set in our apps.

Reply:
I tested it in rails console, I opened a 20 MB text file and appended a null byte to the end of it. This can be an expensive operation if the file is very large.

Didn't know that the limit was 75 GB, thought it was 75 MB. I'll try it with a file of a couple GB

Reply:
**Same test as above, but with a 1GB file**

_When null byte is present_
<img width="561" alt="Screen Shot 2019-11-07 at 3 25 05 PM" src="https://user-images.githubusercontent.com/22248828/68424882-f0701c00-0172-11ea-91c1-6a93f917ac72.png">

_When null byte is not present_
<img width="561" alt="Screen Shot 2019-11-07 at 3 23 22 PM" src="https://user-images.githubusercontent.com/22248828/68424872-eea65880-0172-11ea-9ce9-a5fb8a5c77ff.png">

Reply:
That look good still. Adding the null byte at the end might not be the best scenario, depending on how these search algorithms are implemented. But putting it on the end shall be good enough, doing a quick search it looks like the implementations are very efficient (which of course is expected). Testing other scenarios is overkill, so this is good!